### PR TITLE
feat(query): pretty format for explain

### LINKED ISSUE #5979: Better display format of `EXPLAIN` in new planner
Use terminal graphic characters to link the plan nodes, to make it more readable.

For now, we are using a indent format to display a plan tree, for example:
```
mysql> explain select * from numbers(1) where number = 1;
+----------------------------------------------------------------------------------+
| explain                                                                          |
+----------------------------------------------------------------------------------+
| Project: [number]
    Filter: [number = 1]
        Scan: default.system.numbers
 |
+----------------------------------------------------------------------------------+
1 row in set (0.03 sec)
Read 0 rows, 0.00 B in 0.001 sec., 0 rows/sec., 0.00 B/sec.
```

As the structure of plan tree becomes more complicated, human can hardly read it any more, for example:
```
mysql> explain select     nation,     o_year,     sum(amount) as sum_profit from     (         select             n_name as nation,             extract(year from o_orderdate) as o_year,             l_extendedprice * (1 - l_discount) - ps_supplycost * l_quantity as amount         from             part,             supplier,             lineitem,             partsupp,             orders,             nation         where             s_suppkey = l_suppkey             and ps_suppkey = l_suppkey             and ps_partkey = l_partkey             and p_partkey = l_partkey             and o_orderkey = l_orderkey             and s_nationkey = n_nationkey             and p_name like '%:1%'     ) as profit group by     nation,     o_year order by     nation,     o_year desc;
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| explain                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Project: [n_name,o_year,sum_profit]
    EvalScalar: [sum(amount)]
        Sort: [n_name DESC, o_year DESC]
            Aggregate: group items: [nation, o_year], aggregate functions: [sum(amount)]
                EvalScalar: [amount, nation, o_year]
                    Project: [n_name,o_year,amount]
                        EvalScalar: [-(*(l_extendedprice, -(1, l_discount)), *(ps_supplycost, l_quantity)), toYear(o_orderdate)]
                            HashJoin: INNER, build keys: [n_nationkey], probe keys: [s_nationkey]
                                HashJoin: INNER, build keys: [o_orderkey], probe keys: [l_orderkey]
                                    HashJoin: INNER, build keys: [ps_suppkey, ps_partkey], probe keys: [l_suppkey, l_partkey]
                                        HashJoin: INNER, build keys: [l_suppkey, l_partkey], probe keys: [s_suppkey, p_partkey]
                                            CrossJoin
                                                Filter: [LIKE(p_name, %:1%)]
                                                    Scan: default.tpch.part
                                                Scan: default.tpch.supplier
                                            Scan: default.tpch.lineitem
                                        Scan: default.tpch.partsupp
                                    Scan: default.tpch.orders
                                Scan: default.tpch.nation
 |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.06 sec)
Read 0 rows, 0.00 B in 0.001 sec., 0 rows/sec., 0.00 B/sec.
```

It will be sweet if we can have a format as CockroachDB:
```
defaultdb=> explain select * from test t, test t1 where t.a = 1;
                                       info
-----------------------------------------------------------------------------------
 distribution: local
 vectorized: true

 • cross join
 │ estimated row count: 3
 │
 ├── • scan
 │     estimated row count: 3 (100% of the table; stats collected 82 days ago)
 │     table: test@primary
 │     spans: FULL SCAN
 │
 └── • filter
     │ estimated row count: 1
     │ filter: a = 1
     │
     └── • scan
           estimated row count: 3 (100% of the table; stats collected 82 days ago)
           table: test@primary
           spans: FULL SCAN
(19 rows)

defaultdb=> explain(opt) select * from test t, test t1 where t.a = 1;
            info
----------------------------
 inner-join (cross)
  ├── scan test [as=t1]
  ├── select
  │    ├── scan test [as=t]
  │    └── filters
  │         └── t.a = 1
  └── filters (true)
(7 rows)
```

### PR DESCRIPTION
**Pull request by jiaoew1991**
Signed-off-by: Enwei Jiao <enwei.jiao@zilliz.com>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

support CockroachDB liked explain format

Fixes #5979

### CODE REVIEW

File: query/src/sql/planner/format/indent_format.rs [hunk 1]
```diff
@@ -37,4 +37,25 @@ where T: Display
...
+        } else if is_last {
+            writeln!(f, "{}└── {}", "│   ".repeat(indent - 1), &self.payload).unwrap()
+        } else {
+            writeln!(f, "{}├── {}", "│   ".repeat(indent - 1), &self.payload).unwrap()
+        }
+        for (i, child) in self.children.iter().enumerate() {
+            child.format_pretty_impl(indent + 1, i == self.children.len() - 1, f)?;
+        }
+        Ok(())
+    }
```
Comment (leiysky) [inline query/src/sql/planner/format/indent_format.rs:60]: I think it's better to create a new file `pretty_format.rs` for this.

**Comment by leiysky** [comment]
Rest LGTM.

File: query/src/sql/planner/format/indent_format.rs (hunk 1 above)
Comment (jiaoew1991) [reply query/src/sql/planner/format/indent_format.rs]: > I think it's better to create a new file `pretty_format.rs` for this.

fixed it😄

**Comment by BohuTANG** [comment]
Hi, there is a test failure:
```
20_0001_planner_v2:                                                     [ FAIL ] - result differs with:
--- /workspace/tests/suites/0_stateless/20+_others/20_0001_planner_v2.result	2022-07-13 02:18:12.545147220 +0000
+++ /workspace/tests/suites/0_stateless/20+_others/20_0001_planner_v2.stdout	2022-07-13 02:23:42.577675575 +0000
@@ -323,15 +323,15 @@
 9
 ===Explain===
 Project: [a]
-    Filter: [t1.a > 0]
-        Scan: default.default.t1
+└── Filter: [t1.a > 0]
+    └── Scan: default.default.t1
 Filter: [(t1.a > 3) OR ((t2.a > 5) AND (t1.a > 1))]
-    HashJoin: INNER, build keys: [t2.a], probe keys: [t1.a], join filters: []
-        Scan: default.default.t1
-        Scan: default.default.t2
+└── HashJoin: INNER, build keys: [t2.a], probe keys: [t1.a], join filters: []
+    ├── Scan: default.default.t1
+    └── Scan: default.default.t2
 HashJoin: INNER, build keys: [t2.a], probe keys: [t1.a], join filters: []
-    Scan: default.default.t1
-    Scan: default.default.t2
+├── Scan: default.default.t1
+└── Scan: default.default.t2
 ===Explain Pipeline===
 Project × 1 processor
   TransformHashJoin × 1 processor
```

https://github.com/datafuselabs/databend/runs/7313362524?check_suite_focus=true#step:3:464

**Comment by Xuanwo** [comment]
Clippy is not happy:

```shell
--> query/src/sql/planner/format/pretty_format.rs:53:61
|
53|                (child_prefix.clone() + LAST_REGULAR_PREFIX).clone(),
| ^^^^^^^^help: remove this
|
= note: `-D clippy::redundant-clone` implied by `-D warnings`
note: this value is dropped without further use
--> query/src/sql/planner/format/pretty_format.rs:53:17
|
53|                (child_prefix.clone() + LAST_REGULAR_PREFIX).clone(),
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
= help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
error: redundant clone
--> query/src/sql/planner/format/pretty_format.rs:54:51
|
54|                (child_prefix + LAST_CHILD_PREFIX).clone(),
| ^^^^^^^^help: remove this
|
note: this value is dropped without further use
--> query/src/sql/planner/format/pretty_format.rs:54:17
|
54|                (child_prefix + LAST_CHILD_PREFIX).clone(),
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
= help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#redundant_clone
```

**Comment by Xuanwo** [comment]
@mergify update

**Comment by Xuanwo** [comment]
cc @leiysky PTAL

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**